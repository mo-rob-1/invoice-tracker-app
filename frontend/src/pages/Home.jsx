import { Link } from "react-router-dom";

function Home() {
  return (
    <>
      <section className="heading">
        <h1>Track Your Invoices</h1>
        <p>Please choose from an option below</p>
      </section>

      <Link to="/new-invoice" className="btn btn-reverse btn-block" data-testid="new-invoice-link">
        Create New Invoice
      </Link>

      <Link to="/invoices" className="btn  btn-block" data-testid="view-invoices-link">
        View My Invoices
      </Link>
    </>
  );
}

export default Home;
