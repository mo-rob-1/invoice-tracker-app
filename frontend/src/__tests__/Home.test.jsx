import { render, screen } from "@testing-library/react";
import Home from "../pages/Home";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";

test("renders home page", () => {
  const history = createMemoryHistory();
  const { asFragment } = render(
    <Provider store={store} history={history}>
      <Router>
        <Home />
      </Router>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();

  expect(screen.getByText("Track Your Invoices")).toBeInTheDocument();
  expect(screen.getByText("Please choose from an option below")).toBeInTheDocument();
});

it("renders the 'create new invoice' link", () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <Router>
        <Home />
      </Router>
    </Provider>
  );

  const link = getByTestId("new-invoice-link");
  expect(link).toBeTruthy();
});

it("renders the 'view my invoices' link", () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <Router>
        <Home />
      </Router>
    </Provider>
  );

  const link = getByTestId("view-invoices-link");
  expect(link).toBeTruthy();
});
