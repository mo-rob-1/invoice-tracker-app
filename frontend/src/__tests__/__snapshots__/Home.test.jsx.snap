// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page 1`] = `
<DocumentFragment>
  <section
    class="heading"
  >
    <h1>
      Track Your Invoices
    </h1>
    <p>
      Please choose from an option below
    </p>
  </section>
  <a
    class="btn btn-reverse btn-block"
    data-testid="new-invoice-link"
    href="/new-invoice"
  >
    Create New Invoice
  </a>
  <a
    class="btn  btn-block"
    data-testid="view-invoices-link"
    href="/invoices"
  >
    View My Invoices
  </a>
</DocumentFragment>
`;
